### PR TITLE
Setup/process data in model

### PR DESCRIPTION
**Pull request by M-AminAlizadeh**
The things that I did in this PR:

- Create models for all entities
- Set up associations between models
- Create custom methods for models

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :post, foreign_key: 'author_id'
+  has_many :comment
+  has_many :like
+
```
Comment (BenMukebo) [inline app/models/user.rb:5]: - [ ] Great job @M-AminAlizadeh! 👏🏻 🥇 However, please note that Ruby on Rails is an intelligent framework that places great emphasis on naming conventions to determine how the mapping between models and database tables should be created. 🤔 

- When using `has_many` association types, ensure that the model or table name is pluralized. For instance:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments
has_many :likes
```
- Conversely, when utilizing `belongs_to` association types, ensure that the model or table name is singular. Keep up the good work!

> => let's correct this 🔧 ✍🏻

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  has_many :comment
+  has_many :like
+  belongs_to :author, class_name: 'User'
```
Comment (BenMukebo) [inline app/models/post.rb:4]: - [ ] The same things here for the ` has_many :comment` and `has_many :like`. Please correct the association/. 🔧 ✍🏻

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  def display_three_recent_posts
+    Post.where(author: self).order(created_at: :desc).first(3)
+  end
```
Comment (BenMukebo) [inline app/models/user.rb:8]: - [OPTIONAL ] Like an optional suggestion for the user model method that returns the 3 most recent posts. could also be achieved using this simple and straightforward approach.

```ruby
# app/models/user.rb
...

def recent_post
  posts.order('created_at Desc').limit(3)
end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @M-AminAlizadeh,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Has `User`, `Post`, `Comment` and `Like` models inside `app/models/` directory. ✔️ 
- Descriptive README file.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (M-AminAlizadeh) [reply app/models/user.rb:5]: @BenMukebo Nice catch

File: app/models/user.rb (hunk 3 above)
Comment (M-AminAlizadeh) [reply app/models/user.rb:8]: @BenMukebo Better syntax
I changed it

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @M-AminAlizadeh**, 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Good commits history ✔️  
- Correctly used [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_counter_methods`
- The associations between models are set up correctly. keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**